CorsixTH/Src/th_gfx_font.cpp

```diff
@@ -262,20 +262,22 @@ FT_Error freetype_font::set_face(const uint8_t* pData, size_t iLength) {
   return iError;
 }
 
-FT_Error freetype_font::match_bitmap_font(
-    sprite_sheet* pBitmapFontSpriteSheet) {
-  if (pBitmapFontSpriteSheet == nullptr) return FT_Err_Invalid_Argument;
+FT_Error freetype_font::match_bitmap_font(sprite_sheet* font_spritesheet,
+                                          argb_colour colour) {
+  if (font_spritesheet == nullptr) return FT_Err_Invalid_Argument;
 
   // Try to take the size and colour of a standard character (em is generally
   // the standard font character, but for fonts which only have numbers, zero
   // seems like the next best choice).
   for (const char* sCharToTry = "M0"; *sCharToTry; ++sCharToTry) {
     int iWidth;
     int iHeight;
+    argb_colour sheet_colour = 0;
     unsigned int iSprite = *sCharToTry - 31;
-    if (pBitmapFontSpriteSheet->get_sprite_size(iSprite, &iWidth, &iHeight) &&
-        pBitmapFontSpriteSheet->get_sprite_average_colour(iSprite, &colour) &&
+    if (font_spritesheet->get_sprite_size(iSprite, &iWidth, &iHeight) &&
+        font_spritesheet->get_sprite_average_colour(iSprite, &sheet_colour) &&
         iWidth > 1 && iHeight > 1) {
+      font_colour = ((colour & 0xFF000000) == 0) ? sheet_colour : colour;
```

Comment:
Any fully transparent colour feels unnecessary when the default is just 0

Reply:
Only recognize `0` now.